ninja/operation.py

```diff
@@ -31,6 +31,28 @@
 __all__ = ["Operation", "PathView", "ResponseObject"]
 
 
+def load_request(request: HttpRequest):
+    """
+    Loads a non-json PUT/PATCH request as a POST, reading data from request body
+    and placing them in request.POST and request.FILES.
+    This is required to fix the behaviour when we have a PUT/PATCH endpoint that also
+    receives a File as parameter, forcing the data to be sent as multipart/form-data and
+    making the ParamModel look up at request.POST and request.FILES
+    """
+    if request.method in ("PUT", "PATCH") and request.content_type != "application/json":
+        original_method = request.method
+        if hasattr(request, '_post'):
+            del request._post
+            del request._files
+        try:
+            request.method = "POST"
+            request.META['REQUEST_METHOD'] = 'POST'
+            request._load_post_and_files()
+            request.META['REQUEST_METHOD'] = original_method
+            request.method = original_method
+        except Exception:
+            pass
```

Comment:
If we do this.. why not use `contextlib.supress` ?

Reply:
This is more than just a style fix too. `request.META['REQUEST_METHOD']` and `request.method` should be returned to their original method even if the file loading method fails (silently).

Reply:
```python
with contextlib.supress(Exception):
    request.method = "POST"
    request.META['REQUEST_METHOD'] = 'POST'
    request._load_post_and_files()

request.META['REQUEST_METHOD'] = original_method
request.method = original_method
```

I was thinking something like this

Reply:
Yes, your code suggestion is what I had in mind.
I'm not quite sure if it can actually fail, and maybe we shouldn't be catching and silently swallowing an exception anyway... Yeah, this just seems like a bad idea altogether unless @deniswvieira can say why we need to catch it (and with such a broad catch at that)